### DISCUSSION TITLE
Mensurando Variáveis.

### DISCUSSION THREAD

**Discussion by ArtumosGOC** [codaqui/institucional#190] [Geral]
Pessoal acho muito interessante o endereçamento de memoria de dispositivos eletrônicos em geral, por isso venho aqui compartilhar algo que sei e que pode ser muito útil saber disso.

### Oque é endereçamento de memoria?

- Em resumo ele se refere ao processo de atribuir identificadores únicos, chamados de endereços, a locais específicos na memória de um computador. Esses endereços são utilizados para acessar e manipular dados armazenados na memória.

Links: *(mais aprofundado!!)*
- [Geeksforgeeks](https://www.geeksforgeeks.org/addressing-modes/)
- [Tutorialspoint](https://www.tutorialspoint.com/computer_logical_organization/index.htm)

Vídeos:
- [Memory addressing](https://www.youtube.com/watch?v=F-i9qPOIsoA) *- Apenas Legendado!*
- [Endereçamento de memória](https://www.youtube.com/watch?v=G4om0r-Kgi0)

### Imagem

>![image](https://github.com/codaqui/institucional/assets/63540372/311fdd5f-ca81-4fc4-91f8-f859d81a5d14)
0x**** são endereços de memoria, 0x representa que estão em hexadecimal.

Dentro dos endereços é possível notar a presença de 8 dígitos, no quais correspondem a 0, isso é exatamente o exemplo de tamanho de variável, nesse caso uma variável de 8 bits ou 1 byte.

Essa variável de 8 bits possui '2^8 = (256)' possibilidades de combinações de valores.
- Os valores vão:

  >0 a 255
  -128 a 127

# 
### Oque acontece caso exceda esses 8 bits?

Simples, o valor estoura o limite de possibilidade, oque vai mudar é a percepção que cada linguagem, podem retornar erros ou simplesmente zerar a variável, pois ela só conta os 8 bits. 

De maneira matemática se VALOR > (256) então (VALOR % 256) = Novo valor após o estouro de memoria.

### Imagem

>![image](https://github.com/codaqui/institucional/assets/63540372/2a1acd63-26e4-4ecd-987d-15205deb34f4)

**Comment by endersonmenezes** [comment]
@ArtumosGOC tava vendo as coisas aqui, isso aqui merece um Post, acha não?

**Reply by ArtumosGOC** [reply]
É um bom conteúdo, então sim!
